ledger/simulation/trace.go

```diff
@@ -217,6 +228,18 @@ type ScratchChange struct {
 	NewValue basics.TealValue
 }
 
+// StateOperation represents an operation into an app local/global/box state
+type StateOperation struct {
+	logic.AppStateOperationT
+	logic.AppStateEnum
+	basics.AppIndex
+	Key      string
+	NewValue basics.TealValue
```

Comment:
nit: a comment explaining that this will be the zero value for delete operations might be helpful

Reply:
see [jason's pr comment response](https://github.com/algorand/go-algorand/pull/5659/commits/6217847588a6f7ebd07b6490e0be0118e8d7a271)